electrum_nmc/electrum/names.py

```diff
@@ -131,6 +133,62 @@ def validate_value_length(value: bytes):
     value_length = len(value)
     if value_length > value_length_limit:
         raise BitcoinException('value length {} exceeds limit of {}'.format(value_length, value_length_limit))
+    
+def validate_onion_address(address: str) -> None: # The Address should not have .onion (An implementation to check & remove can be considered)
+    try:
+        # Initialize the bytearray to hold the decoded service ID
+        decoded_service_id = bytearray([0] * V3_ONION_SERVICE_ID_RAW_SIZE)
+
+        # Decode the service ID from base32 into the decoded_service_id bytearray
+        decoded_service_id = base64.b32decode(address.encode(), decoded_service_id)
+
+        # Check decoded service ID has the expected length
+        if len(decoded_service_id) != V3_ONION_SERVICE_ID_RAW_SIZE:
+            raise ValueError("Invalid service ID length")
+
+        # Check the version byte
+        version_byte = decoded_service_id[V3_ONION_SERVICE_ID_VERSION_OFFSET]
+        if version_byte > 0x03:
+            raise ValueError(f"Warning: Unknown version byte {version_byte}")
```

Comment:
I'd suggest a more verbose message. Something like `f"Warning: this is a v{version_byte} onion service, which is newer than the v3 onion services that Electrum-NMC knows how to validate. Please verify that this is what you really meant to enter."`